tests/drop_packets/fanout/mellanox/mellanox_fanout.py

```diff
@@ -1,28 +1,72 @@
 import os
 import pytest
 from ..fanout_base import BaseFanoutHandler
+from tests.common.errors import RunAnsibleModuleFail
+from ansible.inventory.manager import InventoryManager
+from ansible.parsing.dataloader import DataLoader
 
 MAX_OPENFLOW_RULE_ID = 65535
-RUN_ANSIBLE_PLAYBOOK = "cd {ansible_path}; ansible-playbook {playbook} -i lab -l {fanout_host} --extra-vars \"{extra_vars}\" -vvvvv"
+RUN_ANSIBLE_PLAYBOOK = "cd {ansible_path}; ansible-playbook {playbook} -i {inventory} -l {fanout_host} --extra-vars \"{extra_vars}\" -vvvvv"
 # Ansible config files
-LAB_CONNECTION_GRAPH = os.path.normpath((os.path.join(os.path.dirname(__file__), "../../../../ansible/files/lab_connection_graph.xml")))
+LAB_CONNECTION_GRAPH_PATH = os.path.normpath((os.path.join(os.path.dirname(__file__), "../../../../ansible/files")))
 ANSIBLE_ROOT = os.path.normpath((os.path.join(__file__, "../../../../../ansible")))
 # Ansible playbook which executes Jinja template
 ANSIBLE_PLAYBOOK = os.path.join(os.path.dirname(__file__), "exec_template.yml")
 # Jinja template which delete specific openflow rule
 DEL_RULE_TEMPLATE = os.path.join(os.path.dirname(__file__), "mlnx_del_of_rule.j2")
 
 
+def is_mellanox_devices(hwsku):
+    """
+    A helper function to check if a given sku is Mellanox device
+    """
+    hwsku = hwsku.lower()
+    return 'mellanox' in hwsku \
+        or 'msn' in hwsku \
+        or 'mlnx' in hwsku
```

Comment:
Suggest to use variable `mellanox_hwskus` in `ansible/group_vars/sonic/variables` to check if a hwsku is Mellanox switch.

Reply:
It's a good suggestion. But my concern is the ```SKU``` name may vary on different OS. For now, not all fanout switches are  running SONiC.